libavformat/dashdec.c

```diff
@@ -1358,23 +1378,35 @@ static int parse_manifest(AVFormatContext *s, const char *url, AVIOContext *in)
                     attr = attr->next;
                     xmlFree(val);
                 }
-                if ((period_duration_sec) >= (c->period_duration)) {
+
+                // Select newest available period
+                if (period_start_sec >= selected_period_start_sec) {
```

Comment:
We can't use ID since is actually a string.  This will work as long as period start is always included in the tag.  From the manifests I have seen, it should be safe to assume this.